src/runtime/entitlements-manager.js

```diff
@@ -237,33 +237,41 @@ export class EntitlementsManager {
 
   /**
    * Sends a pingback that marks a metering entitlement as used.
-   * @param {!Entitlements} entitlements
+   * @param {!Entitlement|null} entitlement
+   * @param {JsonObject|null|undefined} jwtContents
    */
-  consumeMeter_(entitlements) {
-    const entitlement = entitlements.getEntitlementForThis();
+  consumeMeter_(entitlement, jwtContents) {
     if (!entitlement || entitlement.source !== GOOGLE_METERING_SOURCE) {
       return;
     }
-    // Verify GAA params are present, otherwise bail since the pingback
-    // shouldn't happen on non-metering requests.
-    if (!queryStringHasFreshGaaParams(this.win_.location.search)) {
-      return;
-    }
+
+    // If GAA params are present, include them in the pingback.
+    const gaaToken = queryStringHasFreshGaaParams(this.win_.location.search)
+      ? this.getGaaToken_()
+      : '';
 
     this.deps_
       .eventManager()
       .logSwgEvent(AnalyticsEvent.EVENT_UNLOCKED_BY_METER, false);
 
-    const token = this.getGaaToken_();
+    let entitlementSource = EntitlementSource.GOOGLE_SHOWCASE_METERING_SERVICE;
```

Comment:
this feels kinda klunky.   If there is no gaa token and it's not METERED_BY_GOOGLE we should bail rather than pinging back a showcase meter without gaa params.  I suggest moving the gaa token check abpove in an `else` here and if there is no gaa token bail that way we only ever affirmatively set the `entitlmentSource`